docs/testing/stencil-testrunner/02-config.md

```diff
@@ -0,0 +1,20 @@
+---
+title: Testing Config
+sidebar_label: Config
+description: Testing Config
+slug: /testing-config
+---
+
+# Testing Config
+
+The `testing` config setting in `stencil.config.ts` specifies an object that corresponds to the jest configuration that should be used in your tests. Stencil provides a default configuration, which you likely won't need to edit, however it can be extended with the same configuration options as Jest. See the [Configuring Jest Guide](https://jestjs.io/docs/en/configuration.html) for configuration details.
+
+:::note
+Keep in mind that the usual way of configuring Jest (`package.json` and `jest.config.js`) is not used with the `stencil testing` command. Jest can still be used, but configuring the presets, transpilation and setting up the correct commands must be done by the project.
+:::
+
+Some additional Stencil specific options may be set here as well for configuring the e2e tests:
+
+```tsx reference title=""
+https://github.com/ionic-team/stencil/blob/278c336166e27d329fffb686f2ecb3185d156696/src/declarations/stencil-public-compiler.ts#L1817-L1963
```

Comment:
I don't quite expect us to have an answer to this right now, but want to ask so the team as a whole is thinking about this - how do we maintain this long term? Specifically, if I'm a developer that:
- updates any code highlighted here (so the content has changed and the SHA + line numbers have to change)
- moves this code to another file (so the sha, file path, and perhaps line numbers have to change)

How do I know that this needs to be updated? 

I think this is definitely better than what we have today (a static list that I couldn't tell you when it was last updated without looking at a `git blame`), but do want to start thinking through the next steps here.

Reply:
It will be hard to put an automatic mechanism in place for updating sha's without manually reviewing the code examples. At least this way it becomes an intentional process. I can see how having thousands of such examples will become out of date quickly and updating them manually will become tedious. However IMHO there is generally no need to update the examples until there is a reason to do so.

Reply:
Agreed - I've spent a bit of time thinking about this myself (before I knew of this project) 😆 😉 

I can see a world where we implement some kind of programmatic check against the current SHA and the primary branch (`main`) and see if the content is different, but I think that's outside the scope of this PR. Just want to see if/what others thought and to plant that idea seed.

Reply:
this is amazing!

I agree that pointing to a SHA and pulling lines out of a file is much better than what we do now, and I'd be fine with doing this and maybe putting a ticket into the backlog to consider whether we want to add more tooling around this in the future.

If we had a _lot_ of these in the site we could have something which did something like

1. crawl through the `.md` files to find all such references
2. figure out which version the given `.md` is for (i.e. if in `docs` it's `HEAD`, if it's `versioned_docs/version-v4.11` then it's `4.11.x`, etc)
3. grab the git tag in the Stencil repo for that version and do some git wizardry to determine whether the referenced SHA is still the latest commit which changed the referenced lines up to the point of the version that the doc page is specific to
5. if not, fail a CI check on the PR, so that we know we have to update the lines (I think this could be a manual thing, since the starting line and offset could easily change)

That wouldn't proactively push updates to the stencil site when there's a change in the Stencil repo, but I think that 1) eventual consistency is totally fine here and 2) the stencil site does get rebuilt frequently because of edits and dependency updates and whatnot so I don't think it could ever get too far out of date.

However! It's probably a YAGNI / overengineering thing to even really think about tooling like this, especially when this solution + no tooling at all is still a big upgrade over our current copy-paste based solution

Reply:
Sounds like a good Jedi day project 😉